runtimevar/constantvar/example_test.go

```diff
@@ -91,14 +76,21 @@ func ExampleNewError() {
 }
 
 func Example_openVariableFromURL() {
+	// This example is used in https://gocloud.dev/howto/runtimevar/runtimevar/#local
+
+	// import _ "gocloud.dev/runtimevar/constantvar"
+
 	// runtimevar.OpenVariable creates a *runtimevar.Variable from a URL.
+
+	// Variables set up elsewhere:
 	ctx := context.Background()
+
 	v, err := runtimevar.OpenVariable(ctx, "constant://?val=hello+world&decoder=string")
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer v.Close()
-
+	// Ignore unused variables in example:
```

Comment:
Oh yes, this is a weird one.

I kept it in this one because the test actually runs (due to the `// Output` comments), as opposed to just compiling, and I didn't want to reduce coverage.

The `Ignore unused variables in example` comment is not very exact -- the effect is that it drops everything below it, whether they are unused variables or not.

So, three options:

1. I could copy the original version of this test into the `_test.go` file, and then make this one non-running (and consistent with the other `openFoo` examples, that just open).

2. Modify `gatherexamples` to use a different string for `Ignore unused variables`, something like `Drop everything below here for example`, to make it more clear what's happening.

I have a slight preference for 2. because I kind of like having running examples for the in-memory openers. I would do that in a separate PR.

Reply:
Ok, it took me a while to digest your reply and figure out how these magic examples work...
Your comment says "three options", but I only see two. What's the 3rd?

Re 1: My impression was that we compile/run these examples to ensure that they are valid and useful, not to replace actual testing. So why is it so critical to have extra coverage here?

Re 1: This example is already in a _test.go file, so do you mean make a copy of the function in the same file or do you mean to use it in a different file?
(I wish it were somehow possible to "embed" Example_Open code into a Example_Use functions for the best of both worlds.)

Re 2: +1000 to changing the string to something even more clear like "Drop lines below when using as gocloud.dev example".

Re 2: The two options seem independent to me.

Reply:
I guess there were only 2 options :-). Maybe the 3rd one was "do nothing".

RE 1: You are right, it is mostly about ensuring that the tests are valid and useful (and correct). I think that's important for the portable type examples that show usage of the portable type itself, so many of them use the local/in-memory version (here, `constantvar`) so that they can be run.

RE 1: I meant to copy it somewhere else; it could be as a test, or an example.

RE 2: OK, I will file a bug for that improvement. 

RE 2: It's hard to keep examples consistent across multiple developers and over time. It's fairly clear in my head what they should look like:

-- Examples in drivers should just be of the constructor and the URLOpener; they don't have to run (but can run for in-memory/local versions), and don't have to demonstrate any portable-type functionality.

-- Examples in the portable type should use a local/in-memory instance of the portable type to run, if possible, to ensure that the example is correct.

We can probably do a separate pass across examples after this PR.

Reply:
I agree with your general principles. And yes, it is hard to keep such things consistent. I was just having a hard time reviewing the files because I kept looking for a pattern and could not figure one out. Now that I understand that local examples can actually run, so  are the ones that are different.

In general, I vote for keeping examples on point and copying out anything unrelated to another file as a regular test with proper coverage (if one doesn't exist already). But it makes more sense to do that in a separate PR because if we do this here, we should do the same for all examples/tests.

Reply:
Agree.